pkg/vault/error.go

```diff
@@ -0,0 +1,60 @@
+package vault
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net"
+	"net/url"
+	"os"
+	"strings"
+	"syscall"
+)
+
+// Error is the custom error type for this package
+type Error struct {
+	MessageParts  []string
+	OriginalError error
+}
+
```

Comment:
Might want to add a String() func to this so if someone outputs it in a log things look correct.  https://tour.golang.org/methods/17